src/Traits.h

```diff
@@ -872,6 +898,55 @@ struct Traits<std::unordered_map<KeyT, T, HashT, KeyEqualT, AllocatorT>> : publi
     }
 };
 
+
+namespace {
+
+template <typename ArchiveT, typename MatrixT>
+ArchiveT & SerializeMatrix(ArchiveT &ar, MatrixT const &value) {
+    ar.serialize(static_cast<typename MatrixT::Index>(value.rows()));
+    ar.serialize(static_cast<typename MatrixT::Index>(value.cols()));
+
+    for (typename MatrixT::Index rowId = 0; rowId < value.rows(); ++rowId) {
+        for (typename MatrixT::Index colId = 0; colId < value.cols(); ++colId) {
+            Traits<typename MatrixT::Scalar>::serialize(ar, value(rowId, colId));
+        }   
+    }
+    
+    return ar;
+}
+
+template <typename MatrixT, typename ArchiveT>
+MatrixT DeserializeMatrix(ArchiveT &ar) {

```

Comment:
Do these need to be defined in an anonymous namespace or can then be defined in line below?